src/thinblock.cpp

```diff
@@ -28,6 +28,26 @@ extern CCriticalSection cs_thinblockstats;
 extern CCriticalSection cs_orphancache;
 extern map<uint256, COrphanTx> mapOrphanTransactions GUARDED_BY(cs_orphancache);
 
+string formatInfoUnit(double value)
+{
+    static const char *units[] = {"B", "KB", "MB", "GB", "TB", "PB", "EB"};
+
+    if (value < 0)
+        value = 0.0;
```

Comment:
I'm concerned about mis-displaying negative numbers, even if we don't have them right now.  I think you can make it work by removing the above test and using while ((value < -1000.0 || value > 1000.0) .  Can you try it and extend the tests to negatives?